event-watchers/substrate/src/vanchor_encrypted_output_handler.rs

```
@@ -68,7 +68,7 @@ impl EventHandler<SubstrateConfig> for SubstrateVAnchorEncryptedOutputHandler {
                 .await?
                 .fetch(&next_leaf_index_addr)
                 .await?
-                .unwrap();
+                .expect("fetch next leaf index from storage");
```

Not sure whats the best way to handle this, maybe return an Err?

---

Actually, whenever we fail to fetch a storage value, for whatever reason, we should return with an error. Please do not use `unwrap_or_default` for this case, the default case here is not desirable.